sycl/doc/extensions/proposed/sycl_ext_oneapi_async_memory_alloc.asciidoc

```diff
@@ -0,0 +1,729 @@
+    auto e3 = q.submit_with_event([&](handler &cgh) {
+      cgh.depends_on(e2);
+      async_free(q, temp);
+    });
+  }
+
+  // memory pool releases memory back to the OS
+  q.wait();
+}
+----
+
+Another example of memory pool usage is described in the example below. In this
+example rather than creating an explicit memory pool the default memory pool is
+being used instead. There is also additional queue synchronization between the
+commands enqueued which would ordinarily lead to memory being released back to
+the OS, however, the threshold for the memory pool is extended to the SYCL RT
+will maintain this memory allocation, and therefore still provide the benefit of
+re-allocating memory from the memory pool.
+
+[source,c++]
+----
+int main(int argc, char *argv[])
+{
+  queue q(property::queue::in_order);
+
+  ext::oneapi::experimental::memory_pool memPool
+    = q.get_context().ext_oneapi_get_default_memory_pool(usm::alloc::device);
+
+  memPool.set_new_threshold(1024);
+  
+  {
+    // memory pool allocates memory from the OS
+    void *temp = async_malloc_from_pool(q, 1024, memPool);
+
+    // memory allocation is used for first kernel
+    parallel_for(q, range{1024}, [=](id<1> idx) {
+      do_something(idx, temp);
+    });
+
+    // memory is available to be used by another allocation
+    async_free(q, temp);
+  }
+
+  // memory pool does not release memory back to the OS as it is still within
+  // the specified threshold
+  q.wait();
+
+  {
+    // memory pool re-uses previously allocated memory
+    void *temp = async_malloc_from_pool(q, 1024, memPool);
+
+    // memory allocation is used for second kernel
+    parallel_for(q, range{1024}, [=](id<1> idx) {
+      do_something_else(idx, temp);
+    });
+
+    // memory is available to be used by another allocation
+    async_free(q, temp);
+  }
+
+  // again memory pool does not release memory back to the OS
+  q.wait();
+}
+----
+
+
+== Specification
+
+=== Feature test macro
+
+This extension provides a feature-test macro as described in the core SYCL
+specification.  An implementation supporting this extension must predefine the
+macro `SYCL_EXT_ONEAPI_ASYNC_MEMORY_ALLOC` to one of the values defined in the
+table below.  Applications can test for the existence of this macro to determine
+if the implementation supports this feature, or applications can test the
+macro's value to determine which of the extension's features the implementation
+supports.
+
+[%header,cols="1,5"]
+|===
+|Value
+|Description
+
+|1
+|The APIs of this experimental extension are not versioned, so the
+ feature-test macro always has this value.
+|===
+
+=== Memory pool
+
+This extension introduces the `memory_pool` class, which provides a handle to a
+memory pool owned by the SYCL runtime or a specific backend, and adheres to the
+SYCL common reference semantics.
+
+Memory pools have the following properties:
+
+* An allocation max size (in bytes) is used to manage to total amount of memory
```

Comment:
```suggestion
* An allocation max size (in bytes) is used to manage the total amount of memory
```

Reply:
Fixed in https://github.com/intel/llvm/pull/14800/commits/2c51adf28ece21c0e91c2c0ec85440ae1153e536.